### PR TITLE
style: improve search page

### PR DESCRIPTION
**Pull request by with-heart**
This PR makes a number of improvements to the search page, mostly by improving responsiveness and creating a theme `TagButton` component.

# Screenshots

image
image

# Major Changes

- upgraded `chakra` to `^1.0.0@next` (needed for component theming via `useStyleConfig`)
- moved `theme` to `src/theme` folder (importable via `@theme`)
- created `TagList` + `TagButton` components (tags are now checkboxes under the hood)

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -38,7 +39,9 @@ export default function App({ Component, pageProps }) {
         <CSSReset />
         <Navbar />
         <GlobalStyle />
-        <Component {...pageProps} />
+        <Box mx={[2, 0]}>
```
Comment (with-heart) [inline pages/_app.js:42]: Adds a slight x-margin to mobile pages so they're not scrunched directly against the edge.

File: pages/blogs/search.js [hunk 2]
```diff
@@ -7,28 +7,27 @@ import matter from 'gray-matter';
...
+export default function SearchPage({ allMdx }) {
   const [filteredBlogs, setFilteredBlogs] = React.useState(allMdx);
 
   const handleFilter = (data) => {
     setFilteredBlogs(data);
   };
 
   return (
-    <Flex>
+    <Box pb={3}>
```
Comment (with-heart) [inline pages/blogs/search.js:18]: `Flex` was causing the container to not take up the full width, and it's not needed anyway. Bottom padding makes it so the page doesn't end abruptly at the bottom border of the last content box.

File: pages/blogs/search.js [hunk 3]
```diff
@@ -7,28 +7,27 @@ import matter from 'gray-matter';
...
         <Search blogs={allMdx} handleFilter={handleFilter} />
-        <Flex direction="column" justify="space-evenly" h="80vh">
-          {filteredBlogs &&
-            filteredBlogs.map((blog) => (
-              <ContentBox key={blog.slug} blog={blog} />
-            ))}
-        </Flex>
-      </Flex>
-    </Flex>
+        <Stack spacing={[2, 6, 12]}>
```
Comment (with-heart) [inline pages/blogs/search.js:22]: Responsive spacing makes this look crispier on mobile, which leaving plenty of whitespace on larger screens.

File: src/components/ContentBox.js [hunk 4]
```diff
@@ -1,20 +1,31 @@
...
+    >
+      <Box
+        role="group"
+        maxW="500px"
+        border="1px"
+        borderColor="black"
+        borderRadius="8px"
+        p={8}
+      >
+        <Text _groupHover={{ textDecor: 'underline' }}>{blog.title}</Text>
```
Comment (with-heart) [inline src/components/ContentBox.js:19]: Only the title is underlined on hover now. Looked better to me but it's a stylistic choice 🤷

**Comment by domitriusclark** [comment]
Great changes, but definitely some conflicts due to a bunch of new PR's. 

Going to iron out the process more this weekend. Would you like me to just implement your changes into a new PR instead of you fighting with the conflicts @with-heart

**Comment by with-heart** [comment]
@domitriusclark Conflicts don't scare me 😁 I'll fix them later!

**Comment by domitriusclark** [comment]
Haha I would never doubt your powers

**Comment by with-heart** [comment]
Should be good to go!

**Comment by domitriusclark** [comment]
Thank you so much @with-heart !!

**Pull request merged**